connect/runtime/src/main/java/org/apache/kafka/connect/util/LoggingContext.java

```diff
@@ -0,0 +1,121 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.util;
+
+import org.slf4j.MDC;
+
+/**
+ * A utility for defining Mapped Diagnostic Context (MDC) for SLF4J logs.
+ */
+public final class LoggingContext {
+
+    /**
+     * The parameter keys used by Connect.
+     */
+    public static final class Parameters {
+
+        /**
+         * The name of the Mapped Diagnostic Context (MDC) key that defines the name of a <i>connector</i>.
+         */
+        public static final String CONNECTOR_NAME = "connector.name";
+
+        /**
+         * The name of the Mapped Diagnostic Context (MDC) key that defines the task number of a <i>connector</i>.
+         */
+        public static final String CONNECTOR_TASK = "connector.task";
+
+        /**
+         * The name of the Mapped Diagnostic Context (MDC) key that defines a <i>scope</i> within a connector.
+         */
+        public static final String CONNECTOR_SCOPE = "connector.scope";
+    }
+
+    /**
+     * The {@link Parameters#CONNECTOR_SCOPE scope} values used by Connect.
+     */
+    public enum Scope {
+        /**
+         * The {@link Parameters#CONNECTOR_SCOPE scope} value for the worker as it starts a connector.
+         */
+        WORKER("worker"),
+
+        /**
+         * The {@link Parameters#CONNECTOR_SCOPE scope} value for {@link org.apache.kafka.connect.connector.Task}
+         * implementations.
+         */
+        TASK("task"),
+
+        /**
+         * The {@link Parameters#CONNECTOR_SCOPE scope} value for committing offsets.
+         */
+        OFFSETS("offsets");
+
+        private final String value;
+        Scope(String value) {
+            this.value = value;
+        }
+        public String value() {
+            return this.value;
+        }
+    }
+
+    /**
+     * Modify the current {@link MDC} logging context to set the {@link Parameters#CONNECTOR_NAME connector name} to the
+     * supplied name and the {@link Parameters#CONNECTOR_SCOPE scope} to {@link Scope#WORKER}.
+     *
+     * @param connectorName the connector name; may not be null
+     */
+    public static void forConnector(String connectorName) {
+        MDC.put(Parameters.CONNECTOR_NAME, connectorName);
+        MDC.remove(Parameters.CONNECTOR_TASK);
+        MDC.put(Parameters.CONNECTOR_SCOPE, Scope.WORKER.value());
```

Comment:
actually, i guess i see here that for the connector itself we'd use `WORKER`. why is that?

I know we swap out the context such that some logs would be coming from the worker (e.g. the log messages indicating we are going to instantiate the connector when we start it), but that's also true for tasks where we scope to `task`.

when reading through this initially and i saw worker, I assumed anything logged for that would indicate something that might not invoke the connector/task at all but is done on behalf of them (initially i was thinking offset commit for this but that would invoke; possibly something having to do with configs or statuses?), or where there's not a connector name yet (e.g. validations).

Reply:
I guess because the WorkerTask is doing more than the connector is, but it is doing it for a particular named connector. So when viewing the log, I'd want to see all of the worker-related logic (whether or not it's Connect code or Connector code) in the "worker" context, and all task-related logic (whether it's Connect's WorkerTask or connector task) in the "task" context. The class names in each log message tell me exactly where that code comes from.

Based upon your previous question, is your main question why we use "worker" and not "connector" for the scope? I think I had used "connector" at some point, and think my primary reason was that "connector" scope was used for much more than just what the individual `Connector` instance was doing (even how a user might think of it). Plus "worker" was shorter, which was a small benefit.